sdk/core/azure-core/inc/azure/core/internal/cryptography/sha_hash.hpp

```diff
@@ -17,6 +17,62 @@
 
 namespace Azure { namespace Core { namespace Cryptography { namespace _internal {
 
+  /**
+   * @brief Defines #Sha1Hash.
+   *
+   * @warning SHA1 is a deprecated hashing algorithm and SHOULD NOT be used,
```

Comment:
SHA1 is already in an `_internal` namespace. Why should we move it to _detail? 

If we should move SHA1, it seems to me that we should also move the other SHAxxx functions to `_detail`.

And the reason SHA1 is in core is that SHA1 is used for things other than just websockets.

For instance, in attestation, because it *wasn't* in core, I had to add support for SHA1 by hand inside attestation. If it was in core, I would have used it.

Reply:
IMO, the words `SHOULD NOT be used` should translate to "if are sure you need it, you have to implement it".

Since `sha1` is no longer secure, we would hate to see some security test fail after finding a `sha1` API available.